cmd/node.go

```diff
@@ -0,0 +1,79 @@
+package cmd
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+)
+
+type Node interface {
+	module.ReadyDoneAware
+
+	// Run initiates all common components (logger, database, protocol state etc.)
+	// then starts each component. It also sets up a channel to gracefully shut
+	// down each component if a SIGINT is received.
+	Run()
+
+	// ShutdownSignal returns a channel that is closed when shutdown has commenced.
+	ShutdownSignal() <-chan struct{}
+}
+
+type FlowNodeImp struct {
+	module.ReadyDoneAware
+	*NodeConfig
+	Logger zerolog.Logger
+	sig    chan os.Signal
+}
+
+func NewNode(builder NodeBuilder, cfg *NodeConfig, logger zerolog.Logger, sig chan os.Signal) Node {
+	return &FlowNodeImp{
+		ReadyDoneAware: builder,
```

Comment:
Ah, is this because the actual implementation for the two are still in the same underlying struct?

In this case we don't really need a separate `FlowNodeImp` at all right?

Reply:
yea, that's an artifact of this change being broken into 2 PRs. In #1559 `FlowNodeImp` embeds a `ComponentManager` which moves the `Component` interface.